src/gwt/acesupport/acemode/markdown_highlight_rules.js

```diff
@@ -195,14 +208,51 @@ var MarkdownHighlightRules = function() {
         }, { // h2
             token: "markup.heading.2",
             regex: "^\\-{3,}(?=\\s*$)"
+        }, {
+            // opening fenced div
+            token: "fenced_open", 
+            regex: "^[:]{3,}\\s*.*$", 
+            onMatch: function(val, state, stack) {
+                if (!$rainbowFencedDivs) {
+                    return "keyword.operator";
+                }
+
+                stack = stack || [];
+                stack[0] = state;
+                stack[16] = stack[16] || 0;  // next open color
```

Comment:
Not for this PR, but something to keep in our heads -- it's a little unfortunate that different modes + highlight rules are trying to write into difference indices of this stack, without us really having a way of communicating who gets access to what part of the stack.

I think we should consider a "safer" API for accessing things in the stack, e.g. something like:

```
state = getState(stack, "fenced-divs")
```

and internally `getState()` would retrieve properties from a Javascript object / map stored in the stack at some known / safe location, e.g.

```
function getState(stack, key) {
  var stateData = stack[STATE_DATA_KEY];
  stateData[key] = stateData[key] || {};
  return stateData[key];
```

and `STATE_DATA_KEY` would be some "safe" index into the stack that we feel comfortable storing a property bag. Does that make sense?